app/src/main/java/com/example/tasky/common/data/remote/AuthInterceptor.kt

```diff
@@ -1,20 +1,26 @@
 package com.example.tasky.common.data.remote
 
 import com.example.tasky.common.domain.SessionStateManager
+import com.example.tasky.common.domain.TokenManager
 import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.runBlocking
 import okhttp3.Interceptor
 import okhttp3.Response
 import javax.inject.Inject
 
 class AuthInterceptor @Inject constructor(
-    private val sessionStateManager: SessionStateManager
+    private val sessionStateManager: SessionStateManager,
+    private val tokenManager: TokenManager
 ) : Interceptor {
     companion object {
         const val AUTH = "Authorization"
     }
 
     override fun intercept(chain: Interceptor.Chain): Response {
+        runBlocking {
+            tokenManager.shouldRefreshToken()
+        }
```

Comment:
This code looks like it doesn't do anything.. From a function `shouldRefreshToken()` I would expect that it returns whether to refresh the token, but not actually refreshing it. That would be a side-effect since the function doesn't communicate that.
If you want this function to actually refresh the token if necessary, why not call it like that? Something like `refreshIfNecessary` will be very clear :)